reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/MicrometerObservationListener.java

```diff
@@ -0,0 +1,211 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.observability.micrometer;
+
+import io.micrometer.observation.Observation;
+
+import reactor.core.observability.SignalListener;
+import reactor.core.publisher.SignalType;
+import reactor.util.annotation.Nullable;
+
+/**
+ * A {@link SignalListener} that makes timings using the {@link io.micrometer.observation.Observation} API from Micrometer 1.10.
+ * <p>
+ * This is a transposition of {@link MicrometerMeterListener}, but only retains the timers.
+ *
+ * @author Simon Baslé
+ */
+final class MicrometerObservationListener<T> implements SignalListener<T> {
+
+	static final String OBSERVATION_FLOW = ".observation.flow";
+	static final String OBSERVATION_VALUES = ".observation.values";
+
+
+	final MicrometerObservationListenerConfiguration configuration;
+	@Nullable
+	final Observation                                signalIntervalObservation;
+	final Observation                                subscribeToTerminalObservation;
+
+	/**
+	 * Scope is first opened in onSubscribe.
+	 * It is then closed and opened in each onNext.
+	 * Finally, last Scope is closed in either of the terminal states (cancelled, completed, error).
+	 */
+	@Nullable
+	Observation.Scope onNextScope;
+
+	boolean valued;
+
+	MicrometerObservationListener(MicrometerObservationListenerConfiguration configuration) {
+		this.configuration = configuration;
+
+		this.valued = false;
+		if (configuration.isMono) {
+			//for Mono we don't record signalInterval (since there is at most 1 onNext, it's the same as subscribeToTerminalObservation).
+			//Note that we still need a mean to distinguish between empty Mono and valued Mono for recordOnCompleteEmpty
+			signalIntervalObservation = null;
+			onNextScope = Observation.Scope.NOOP;
+		}
+		else {
+			this.signalIntervalObservation = Observation.createNotStarted(
+				configuration.sequenceName + OBSERVATION_VALUES,
+				configuration.registry
+			).lowCardinalityKeyValues(configuration.commonKeyValues);
+		}
+
+		subscribeToTerminalObservation = Observation.createNotStarted(
+			configuration.sequenceName + OBSERVATION_FLOW,
+			configuration.registry
+		).lowCardinalityKeyValues(configuration.commonKeyValues);
+	}
+
+	@Override
+	public void doOnCancel() {
+		//Due to dealing with scopes, we have to stop the current scope.
+		//Note this could skew the onNext counter off by one if the Observation is a facade over metrics.
+		if (this.onNextScope != null) {
+			this.onNextScope.close();
+		}
+
+		Observation observation = subscribeToTerminalObservation
+			.lowCardinalityKeyValue(MicrometerMeterListener.TAG_KEY_STATUS, MicrometerMeterListener.TAG_STATUS_CANCELLED)
+			.highCardinalityKeyValue(MicrometerMeterListener.TAG_KEY_EXCEPTION, "");
```

Comment:
We have an `error` method on the `Observation`. We will then take care of setting tags in the proper handlers.